File: movie-frontend/src/constants/ApiEndpoints.ts
const BASE_URL = import.meta.env.VITE_API_BASE_URL;

export const API_ENDPOINTS = {
    ADMIN: {
        DASHBOARD: `${BASE_URL}/admin/statistics`,
        AI_STATUS: `${BASE_URL}/admin/ai-status`,
        RETRAIN_AI: `${BASE_URL}/admin/retrain-ai`,
        UPDATE_RECOMMENDATIONS: `${BASE_URL}/admin/update-recommendations`,
    },
    AUTH: {
        REGISTER: `${BASE_URL}/auth/register`,
        LOGIN: `${BASE_URL}/auth/login`,
        EXIST_EMAIL: `${BASE_URL}/auth/exist-email`,
        GET_PROFILE: `${BASE_URL}/auth/profile`,
    },
    CATEGORY: {
        GET_ALL: `${BASE_URL}/category/genres`,
    },
    CHATBOT: {
        MESSAGE: `${BASE_URL}/chatbot/message`,
    },
    HOMEPAGE: {
        TRENDING: `${BASE_URL}/movie/trending?limit=10`,
        HERO_MOVIE: `${BASE_URL}/movie/hero`,
        PREFERRED_GENRES: `${BASE_URL}/movie/preferredGenres?limit=10`,
        TOP10: `${BASE_URL}/movie/top10`,
    },
    MOVIE: {
        GENRES: `${BASE_URL}/category/genres`,
        FILTER: `${BASE_URL}/movie/filter`,
    },
    MEDIA_CONTENT: {
        GET_BY_ID: (movieId: number) => `${BASE_URL}/mediacontent/${movieId}`,
    },
    USER: {
        GET_RATE_BY_MOVIE_ID: (movieId: number) => `${BASE_URL}/user-interaction/rate/${movieId}`,
        ADD_RATE: `${BASE_URL}/user-interaction/rate`,
        
        GET_FAVORITES: `${BASE_URL}/user-interaction/favorites`,
        ADD_FAVORITE: `${BASE_URL}/user-interaction/favorites/add`,
        REMOVE_FAVORITE: (movieId: number) => `${BASE_URL}/user-interaction/favorites/delete/${movieId}`,
        
        ADD_WATCH_HISTORY: `${BASE_URL}/user-interaction/addWatchHistory`,
        EXISTS_WATCH_HISTORY: (movieId: number) => `${BASE_URL}/user-interaction/checkWatchHistory/${movieId}`,
    },
    SEARCH: {
        SUGGEST: `${BASE_URL}/search/suggestion`,
        ALL: `${BASE_URL}/search/all`,
    },
    ONBOARDING: {
        POST: `${BASE_URL}/user-interaction/onboarding`,
    },
    RECOMMENDATION: {
        CF: {
            UPDATE_RECOMMENDATIONS: `${BASE_URL}/recommendation/cf/update-recommendations`,
            USER: (userId: number) => `${BASE_URL}/recommendation/cf/user/${userId}`,
            SIMILAR: (movieId: number) => `${BASE_URL}/recommendation/cf/similar/${movieId}`,
        },
        CBF: {
            SEARCH: `${BASE_URL}/recommendation/cbf/search`,
            SIMILAR: (movieId: number) => `${BASE_URL}/recommendation/cbf/similar/${movieId}`,
        }
    }
} as const;

File: movie-backend/src/main/java/nlu/fit/movie_backend/constants/ApiEndpoints.java
package nlu.fit.movie_backend.constants;

public interface ApiEndpoints {
    interface Admin {
        String BASE = "/admin";

        String STATISTICS = BASE + "/statistics";
        String USERS = BASE + "/user/users";
        String DELETE_USER = BASE + "/user/{id}";
        String MOVIES = BASE + "/movie/movies";
        String ADD_MOVIE = BASE + "/movie/movies";
        String UPDATE_MOVIE = BASE + "/movie/movies";
        String DELETE_MOVIE = BASE + "/movie/movies" + "/{id}";
        String AI_STATUS = BASE + "/ai/movies";
        String RETRAIN_AI = BASE + "/ai/update-recommendations";
        String CHECK_DUPLICATE = BASE + "/movie/check-duplicate";
    }

    interface Auth {
        String BASE = "/auth";

        String REGISTER = BASE + "/register";
        String LOGIN = BASE + "/login";
        String PROFILE = BASE + "/profile";
        String EXIST_EMAIL = BASE + "/exist-email";
    }

    interface Chatbot {
        String BASE = "/chatbot";

        String CHATBOT = BASE + "/message";
        String HISTORY = BASE + "/history";
    }

    interface Category {
        String BASE = "/category";

        String ALL = BASE + "/genres";
    }

    interface MediaContent {
        String BASE = "/mediacontent";

        String GET_BY_ID = BASE + "/{movieId}";
    }

    interface Movie {
        String BASE = "/movie";

        String MOVIES = BASE + "/movies";
        String LATEST_MOVIES = BASE + "/latest";
        String TRENDING = BASE + "/trending";
        String HERO = BASE + "/hero";
        String TOP10 = BASE + "/top10";
        String MOVIE_GENRES = BASE + "/preferredGenres";
        String FILTER = BASE + "/filter";
    }

    interface Recommendation {
        String BASE = "/recommendation";

        String GET_CF_USERID = BASE + "/cf/user/{userId}";
        String GET_CF_SIMILAR_BY_MOVIE_ID = BASE + "/cf/similar/{movieId}";

        String GET_CBF_SEARCH = BASE + "/cbf/search";
        String GET_CBF_SIMILAR_BY_MOVIE_ID = BASE + "/cbf/similar/{movieId}";
        String GET_HYBRID_RECOMMENDATION = BASE + "/hybrid/{userId}";

    }

    interface Search {
        String BASE = "/search";
        String SUGGESTION = BASE + "/suggestion";
        String ALL = BASE + "/all";
    }

    interface UserInteraction {
        String BASE = "/user-interaction";

        String GET_ALL_FAVORITE = BASE + "/favorites";
        String ADD_MOVIE_FAVORITE = BASE + "/favorites/add";
        String DELETE_MOVIE_FAVORITE = BASE + "/favorites/delete/{movieId}";

        String ADD_RATE = BASE + "/rate";
        String GET_RATE = BASE + "/rate/{mediaContentId}";

        String ONBOARDING = BASE + "/onboarding";

        String ADD_WATCH_HISTORY = BASE + "/addWatchHistory";
        String EXISTS_WATCH_HISTORY = BASE + "/checkWatchHistory/{mediaContentId}";

    }

}
